quickwit/quickwit-proto/src/search/mod.rs

```diff
@@ -243,3 +245,48 @@ impl PartialHit {
         }
     }
 }
+
+/// Serializes the Split fields.
+///
+/// `fields_metadata` has to be sorted.
+pub fn serialize_split_fields(list_fields: ListFields) -> Vec<u8> {
+    let payload = list_fields.encode_to_vec();
+    let compression_level = 3;
+    let payload_compressed = zstd::stream::encode_all(&mut &payload[..], compression_level)
+        .expect("zstd encoding failed");
+    let mut out = Vec::new();
+    // Write Header -- Format Version 2
+    let format_version = 2u8;
+    out.push(format_version);
+    // Write Payload
+    out.extend_from_slice(&payload_compressed);
+    out
+}
+
+/// Reads a fixed number of bytes into an array and returns the array.
+fn read_exact_array<R: Read, const N: usize>(reader: &mut R) -> io::Result<[u8; N]> {
```

Comment:
nitpick...
`impl Trait` is nice in that kidn of funciton as it removes the need to `::<_, 4>` on call sites.

```suggestion
fn read_exact_array<const N: usize>(reader: &mut impl std::io::Read) -> io::Result<[u8; N]> {
```